### PR TITLE
[pf tool] Support tool init with advance info, such as icon, tags and category

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
https://github.com/microsoft/promptflow/commit/70e65be8d8ee072da9fade86041df4a66df0b280
Support tool init with advance info, such as icon, tags and category

```
pf tool init --package package_tool --tool tool_name --set icon=<icon-path> category=<category> tags="{'key': 'value'}"
```

tool meta:
```
"test_cli_package.cli_tool.cli_tool": {
    "name": "cli_tool",
    "type": "python",
    "inputs": {
      "connection": {
        "type": [
          "CustomConnection"
        ]
      },
      "input_text": {
        "type": [
          "string"
        ]
      }
    },
    "description": "This is cli_tool tool",
    "module": "test_cli_package.cli_tool",
    "function": "cli_tool",
    "icon": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAABcElEQVR4nJVSsUoDQRB9e+dd4p0xESHaHBEsJCEKsbf0C+z0B/wHv8PC3sIvEMFWRAJKFCwsDBpIyEbwyJlj9/bcHQtPiBgMmfLx3sybmceICLOUNRP7R0AErQ19EsxUAZtoiYgYYxMRC4B5fVLHRxcPh8/hZdbmN3scsQCYt256fnr7ctKN7lSSjuIoiqJxtjGm3++P7WBZcHKOPWc0Hu/bzWaz1+sppZRSWus0TUejUavViuMYwBwA2y/Mb9RXl1aLXiDEhxCSc95ut6WUruuWSqXBYMA5r9frvu9bAG6WGzu7Z3tOqcEsAZYmUghRq9WCICgWi0mSVCqVarVaLpczS2FirvtKvV9BvOa8Rd/ztNadTkdKGcfxcDjknGutwzDMLBkABJgUMNuba9ArKjVSSt/3oyjK5/NE5LqubduZIPDY/rrjLhxgYcvLO4Dj/dynUChMfxzR98Q/P2YsExBgCDY0GJuarsnR+KdmTusX7Qa9XJS8qLcAAAAASUVORK5CYII=",
    "category": "test_category",
    "tags": {
      "key": "value"
    },
    "package": "test-cli-package",
    "package_version": "0.0.1"
  }
}
```

Folder structure
```
TOOL_WITH_EXTRA_INFO
│   MANIFEST.in
│   README.md
│   setup.py
│
├───icon
│       logo.jpg
│
└───tool_with_extra_info
        func_name.py
        utils.py
        __init__.py

```

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by zhengfeiwang — APPROVED** [review]

File: src/promptflow/setup.py [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "opencensus-ext-azure<2.0.0",  # configure opencensus to send telemetry to azure monitor
     "ruamel.yaml>=0.17.35,<0.18.0",  # used to generate connection templates with preserved comments
     "pyarrow>=13.0.0,<14.0.0",  # used to read parquet file with pandas.read_parquet
+    "pillow",  # used to generate icon data URI for package tool
```
Comment (wangchao1230) [inline src/promptflow/setup.py:50]: shall we have a minimal version and a range here?

File: src/promptflow/promptflow/_cli/data/package_tool/MANIFEST.in.jinja2 [hunk 2]
```diff
@@ -0,0 +1 @@
+include {{ package_name }}/icons
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/data/package_tool/MANIFEST.in.jinja2:1]: why do we need this file?

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/_cli/data/package_tool/MANIFEST.in.jinja2 (hunk 2 above)
Comment (lalala123123) [reply src/promptflow/promptflow/_cli/data/package_tool/MANIFEST.in.jinja2:1]: If the user needs to release this package, those non-Python files need to be include in the package

File: src/promptflow/setup.py (hunk 1 above)
Comment (wangchao1230) [reply src/promptflow/setup.py:50]: if by design, we can keep it.

Comment (lalala123123) [reply src/promptflow/setup.py]: Pillow's version is locked in the range 10.1.0-11.0.0

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**